src/cargo/ops/cargo_new.rs

```diff
@@ -253,8 +253,7 @@ fn check_name(
     if restricted_names::is_non_ascii_name(name) {
         shell.warn(format!(
             "the name `{}` contains non-ASCII characters\n\
-            Support for non-ASCII crate names is experimental and only valid \
-            on the nightly toolchain.",
+            Non-ASCII crate names are not supported by Rust.",
```

Comment:
I think this should say `cargo` not `Rust` since it is emitted by cargo
```suggestion
            Non-ASCII crate names are not supported by cargo.",
```

Reply:
I had also considered that but currently Cargo does support non-ASCII crates without much issue. Its just `rustc` that doesn't. For example, this works fine with `cargo check`:
```
[dependencies]
"ああああ" = { path = "../ああああ" }
```

Reply:
I am not sure whether it's too much, but I might put something like _"it will become a hard error in a future release"_ if we are trying to scare away folks using non-ASCII crate name 😆.

Either way, they will evenutually figure out it won't work when using as an extern crate (dependency).

Reply:
Personally, I wouldn't mind if Cargo still supported it so that you could make binaries with non-ASCII names, regardless of how well everything else on the system handles it :D